src/AppKit/NSFileTypeForHFSTypeCode.cs

```diff
@@ -0,0 +1,38 @@
+//
+// NSFileTypeForHFSTypeCode.cs
+//
+// Author:
+//  Whitney Schmidt <whschm@microsoft.com>
+//
+// Copyright 2019 Microsoft Corporation. All right reserved.
+
+
+using System;
+using System.Runtime.InteropServices;
+
+using Foundation;
+using ObjCRuntime;
+
+namespace AppKit
+{
+
+    public static class NSFileTypeForHFSType {
+
+        [DllImport ("/System/Library/Frameworks/Foundation.framework/Foundation")]
+        extern static IntPtr NSFileTypeForHFSTypeCode (int /* OSType = int32_t */ hfsFileTypeCode);
+
+        [DllImport ("/System/Library/Frameworks/Foundation.framework/Foundation")]
+        extern static int UTGetOSTypeFromString (IntPtr str);
+
+        public static string GetHFSTypeCodeString (string type)
+        {
+            int code;
+            using (NSString ns = (NSString)type)
+                code = UTGetOSTypeFromString (ns.Handle);
+
+            using (var s = Runtime.GetNSObject<NSString> (NSFileTypeForHFSTypeCode (code))) {
+                return (string)s;
+            }
```

Comment:
There are simpler approach, e.g. you can typecast a `NSString` into `string`
But in this case it would be even simpler to write the logic in C# (we already have it in our tests).
Still I don't think it's the right way to expose the values (because `string` are not the best type for constants).